tests/test_model_trainers.py

```diff
@@ -218,3 +223,98 @@ def test_n_jobs_not_new_model():
                 'select model_parameters from results.model_groups'
             ):
                 assert 'n_jobs' not in row[0]
+
+
+class RetryTest(unittest.TestCase):
+    def test_retry_max(self):
+        grid_config = {
+            'sklearn.ensemble.AdaBoostClassifier': {
+                'n_estimators': [10]
+            },
+        }
+
+        engine = None
+        trainer = None
+        with testing.postgresql.Postgresql() as postgresql:
+            engine = create_engine(postgresql.url())
+            ensure_db(engine)
+            trainer = ModelTrainer(
+                project_path='econ-dev/inspections',
+                experiment_hash=None,
+                model_storage_engine=InMemoryModelStorageEngine(project_path=''),
+                db_engine=engine,
+                model_group_keys=['label_name', 'label_window']
+            )
+
+            matrix = pandas.DataFrame.from_dict({
+                'entity_id': [1, 2],
+                'feature_one': [3, 4],
+                'feature_two': [5, 6],
+                'label': ['good', 'bad']
+            })
+            matrix_store = InMemoryMatrixStore(matrix, {
+                'label_window': '1d',
+                'end_time': datetime.datetime.now(),
+                'beginning_of_time': datetime.date(2012, 12, 20),
+                'label_name': 'label',
+                'metta-uuid': '1234',
+                'feature_names': ['ft1', 'ft2']
+            })
+        with patch('time.sleep') as time_mock:
+            with self.assertRaises(sqlalchemy.exc.OperationalError):
+                trainer.train_models(grid_config, dict(), matrix_store)
+            assert len(time_mock.mock_calls) == 13
```

Comment:
I like the 'at least 5 times' idea, because part of the reason we're testing this is to make sure we're configuring the retrying module in a sane way.